packages/query/src/block-processor.ts

```diff
@@ -263,11 +263,11 @@ export class BlockProcessor implements BlockProcessorInterface {
 
     // flushing is slow, avoid it until
     // - wasm says
-    // - every 1000th block
+    // - every 5000th block
     // - every block at tip
     const flushReasons = {
       scannerWantsFlush,
-      interval: compactBlock.height % 1000n === 0n,
+      interval: compactBlock.height % 5000n === 0n,
```

Comment:
Increasing the chunk size for flush operations improves performance.

Reply:
Of all the changes in the PR, is it not this change that represents the nearlly all of the sync speed improvement?

Reply:
> This is something worth thinking about. Why not make the interval say 10k blocks or even bigger? I wonder what is the limiting principle for this.

the trade-off here is between memory pressure and disk I/O pressure. The longer the block processor waits to flush, the more updates accumulate in memory, but less frequent writes leading to lower I/O costs. 

In my testing, I there was a noticeable improvement when increasing the interval from 1k to 5k, but only a marginal change from 5k to 10k.

Reply:
> Also, we sync on `scannerWantsFlush`. This will be true when the fvk successfully decrypts something from a block. What if we don't do that though. Would it be bad if the user just waits until the next interval? Wonder if that would save time as well.

I had the same thought, and it was noticeably more performant, but it resulted in SCT state corruption.

Reply:
> Of all the changes in the PR, is it not this change that represents the nearlly all of the sync speed improvement?

Increasing the interval improved performance on its own, but combining it with these other changes enhanced it even further.

Reply:
> I had the same thought, and it was noticeably more performant, but resulted in SCT state corruption. I can investigate further.

That feels like it should not be the case 🤔 . Can you create an issue to follow up investigating why we can't flush on intervals but are required to flush on finding new notes/swaps?